### PR TITLE
added split pane options to settings

### LINKED ISSUE #79665: Option to not automatically resize all panes when splitting a pane
Right now when you split a pane, all panes are resized so they have equal width:

resize-panes

Personally I'd much rather it preserve the existing widths, and only split the width of the pane that I split. An option for this would be nice.

(Annoyance being that I have two or three columns already; when I click split, I now have to re-resize three or four columns to put them back where I had them.)

### PR DESCRIPTION
**Pull request by dhairyanadapara**
This PR fixes #79665

* Added the setting for splitting the pane into multiple parts
* User has 2 options:
  * Split: Selected editor splits into equal parts
  * Distribute: All editors splits into equal parts
* Same while closing editors

### CODE REVIEW

**Comment by bpasero** [comment]
@dhairyanadapara I think you are adding the option on the wrong layer (monaco editor), shouldn't the setting appear very similar to other existing editor settings, such as [`workbench.editor.closeEmptyGroups`](https://github.com/microsoft/vscode/blob/82ca6ba87f450245f665ab7a5f3a686e2cd00045/src/vs/workbench/browser/workbench.contribution.ts#L100)?

**Comment by dhairyanadapara** [comment]
Thanks, @bpasero. Found out the same. I have moved options to `workbench.editor`

File: src/vs/workbench/browser/parts/editor/editor.ts [hunk 1]
```diff
@@ -37,7 +37,8 @@ export const DEFAULT_EDITOR_PART_OPTIONS: IEditorPartOptions = {
 	openSideBySideDirection: 'right',
 	closeEmptyGroups: true,
 	labelFormat: 'default',
-	iconTheme: 'vs-seti'
+	iconTheme: 'vs-seti',
+	splitSizing: 'Distribute'
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editor.ts]: I would prefer lowercase `distribute` and `split`

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 2]
```diff
@@ -60,6 +60,16 @@ import { isMacintosh, isWindows, isLinux, isWeb, isNative } from 'vs/base/common
...
 			},
+			'workbench.editor.splitSizing': {
+				'type': 'string',
+				'enum': ['Distribute', 'Split'],
+				'default': 'Distribute',
+				'enumDescriptions': [
+					nls.localize('workbench.editor.splitSizing.Distribute', "Splits all the editors to equal parts."),
+					nls.localize('workbench.editor.splitSizing.Split', "Splits the active editor to equal parts.")
+				],
+				'description': nls.localize({ comment: ['This is the description for a setting. Values surrounded by single quotes are not to be translated.'], key: 'splitSizing' }, "Controls the sizing of editor tabs.")
```
Comment (bpasero) [inline src/vs/workbench/browser/workbench.contribution.ts]: `Controls the sizing of editor tabs.` => `Controls the sizing of editors when splitting them.`

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 3]
```diff
@@ -60,6 +60,16 @@ import { isMacintosh, isWindows, isLinux, isWeb, isNative } from 'vs/base/common
 				],
 				'description': nls.localize({ comment: ['This is the description for a setting. Values surrounded by single quotes are not to be translated.'], key: 'tabSizing' }, "Controls the sizing of editor tabs.")
 			},
+			'workbench.editor.splitSizing': {
+				'type': 'string',
+				'enum': ['Distribute', 'Split'],
```
Comment (bpasero) [inline src/vs/workbench/browser/workbench.contribution.ts]: I would prefer lowercase `distribute` and `split`

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 4]
```diff
@@ -60,6 +60,16 @@ import { isMacintosh, isWindows, isLinux, isWeb, isNative } from 'vs/base/common
...
+				'default': 'Distribute',
+				'enumDescriptions': [
+					nls.localize('workbench.editor.splitSizing.Distribute', "Splits all the editors to equal parts."),
```
Comment (bpasero) [inline src/vs/workbench/browser/workbench.contribution.ts]: `workbench.editor.splitSizing.Distribute` => `workbench.editor.splitSizingDistribute`

`workbench.editor.splitSizing.Split` => `workbench.editor.splitSizingSplit`

File: src/vs/workbench/common/editor.ts [hunk 5]
```diff
@@ -1025,6 +1025,7 @@ interface IEditorPartConfiguration {
 	mouseBackForwardToNavigate?: boolean;
 	labelFormat?: 'default' | 'short' | 'medium' | 'long';
 	restoreViewState?: boolean;
+	splitSizing?: 'Split' | 'Distribute';
```
Comment (bpasero) [inline src/vs/workbench/common/editor.ts]: I would prefer lowercase `distribute` and `split`

File: src/vs/workbench/browser/parts/editor/editorPart.ts [hunk 6]
```diff
@@ -497,13 +497,18 @@ export class EditorPart extends Part implements IEditorGroupsService, IEditorGro
 		return group;
 	}
 
+	private _getSizingStyle(): Sizing {
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorPart.ts]:
```suggestion
	private getSplitSizingStyle(): Sizing {
	const splitStyle = this._partOptions.splitSizing;

	if (splitStyle === 'Split') {
		return Sizing.Split;
	}

	return Sizing.Distribute;
}
```

**Review by bpasero — CHANGES_REQUESTED** [review]

**Comment by bpasero** [comment]
Thanks 👍

**Comment by dhairyanadapara** [comment]
Thanks, @bpasero. This was my first contribution to open source and I enjoyed it. If you found any other issue in which I can help, please assign it. Thanks again for helping. :)

**Pull request merged**